packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -182,7 +191,8 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
     _drag = HorizontalDragGestureRecognizer()
       ..onStart = _handleDragStart
       ..onUpdate = _handleDragUpdate
-      ..onEnd = _handleDragEnd;
+      ..onEnd = _handleDragEnd
+      ..dragStartBehavior  = dragStartBehavior;
```

Comment:
this needs to handle the case where it changes dynamically.

(that will also need a test)